.github/workflows/codeql.yml

```diff
@@ -0,0 +1,51 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+name: "CodeQL Advanced"
+
+on:
+  push:
+    branches: [ "main", "release/*" ]
+  pull_request:
+    branches: [ "main", "release/*" ]
```

Comment:
If I'm not mistaken, with these trigger configurations, a commit pushed to an open pull request will trigger two pipelines, one from `push` and the other from `pull_request`. Or is this to accommodate for PR's from forks as they can't push directly to the repository?

Reply:
Won't both only be triggered if a `pull_request` originates from a `push`-allowed branch?

Reply:
Yes, I believe so. Only pushes from the repo will trigger the `push` workflow, otherwise only `pull_request`.